### PR TITLE
Implement mock of httpsnapp

### PR DESCRIPTION
**Pull request by MartinMinkov**
In its current state, this PR mocks out the functionality for the Micro library `httpsnapper` specified in https://github.com/MinaProtocol/mina/issues/9306. It currently exports a `verify` function that returns an expected response from an HTTPS Snapps Oracle. The current mock was manually tested to be a correct input to the `verifyMessage` function in the `client-sdk`. 

Checklist:

- [x] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [x] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose

### CODE REVIEW

**Review by bkase — COMMENTED** [review]
Can you adjust based on latest feedback to the spec? https://github.com/MinaProtocol/mina/pull/9312/files

File: frontend/snapps/httpsnapp/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+type QueryParams = {};
+
+type Response = {
+  ok?: Signed<string>;
+  error?: Error;
+};
+
+/**
+ * Currently, WIP. Sends either a GET or POST request to the HTTPSnapps Oracle API which will
```
Comment (bkase) [inline frontend/snapps/httpsnapp/src/index.ts:23]: is it still WIP?

File: frontend/snapps/httpsnapp/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
+import fetch from 'node-fetch';
```
Comment (bkase) [inline frontend/snapps/httpsnapp/src/index.ts:1]: can you make sure this works in the browser and NodeJS?

File: frontend/snapps/httpsnapp/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  url: string,
+  params: QueryParams,
+  _postData: string
+): Promise<Response> => {
+  // Convert QueryParams to query string
+  const qs = Object.keys(params)
+    .map((key) => `${key}=${params[key]}`)
+    .join('&');
+
+  // Temporary URL, Oracle API runs on localhost for now
```
Comment (bkase) [inline frontend/snapps/httpsnapp/src/index.ts:43]: seems like we should adjust the API to take the URI prefix. I don't know what's the typical way to do this these days in JS-land.

We could do

```js
import HttpSnapps from 'httpsnapps'
const verify = HttpSnapps({ url: "https://localhost:3000" }).verify
```

or

```js
import HttpSnapps from 'httpsnapps'
const verify = HttpSnapps.makeVerify({ url: "https://localhost:3000" })
```

or something else?

File: frontend/snapps/httpsnapp/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+      : `http://localhost:3000?url=${url}&${qs}`;
+
+  let response;
+  if (method === 'GET') {
+    response = await fetch(minaOracle, { method: 'GET' }).catch((err) => {
+      return {
+        error: err,
+      };
+    });
+  } else if (method === 'POST') {
```
Comment (bkase) [inline frontend/snapps/httpsnapp/src/index.ts:56]: can you `throw` if postData is not-nully for now (until we use it)?

**Review by bkase — CHANGES_REQUESTED** [review]
Awesome!

File: frontend/snapps/httpsnapp/src/index.ts (hunk 3 above)
Comment (jasongitmail) [reply frontend/snapps/httpsnapp/src/index.ts:43]: Something like this is probably most common, if you don't expect the params to change:

```js
import HttpSnapps from 'httpsnapps'
const verify = HttpSnapps.verify("https://localhost:3000")
```

or the object param works too:

```js
import HttpSnapps from 'httpsnapps'
const verify = HttpSnapps.verify({ url: "https://localhost:3000" })
```

(It's easy to switch between the two without a breaking change if needed in either case, by checking the param type.)

File: frontend/snapps/httpsnapp/src/index.ts [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+ * @returns
+ */
+export const verify = async (
+  method: Method,
+  url: string,
+  params: QueryParams,
+  _postData: string
+): Promise<Response> => {
+  // Convert QueryParams to query string
+  const qs = Object.keys(params)
```
Comment (jasongitmail) [inline frontend/snapps/httpsnapp/src/index.ts:39]: We could consider [this method](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/URLSearchParams):
```js
const qs = new URLSearchParams(params);
```

File: frontend/snapps/httpsnapp/src/index.ts (hunk 1 above)
Comment (MartinMinkov) [reply frontend/snapps/httpsnapp/src/index.ts:23]: Will fix this before merging!

File: frontend/snapps/httpsnapp/src/index.ts (hunk 2 above)
Comment (MartinMinkov) [reply frontend/snapps/httpsnapp/src/index.ts]: Good catch, I switched out `node-fetch` to use `axios` instead which should work in both the browser and NodeJS.

File: frontend/snapps/httpsnapp/src/index.ts (hunk 4 above)
Comment (MartinMinkov) [reply frontend/snapps/httpsnapp/src/index.ts]: Done!

File: frontend/snapps/httpsnapp/src/index.ts (hunk 5 above)
Comment (MartinMinkov) [reply frontend/snapps/httpsnapp/src/index.ts]: Good catch switched it out. Thanks!

File: frontend/snapps/httpsnapp/src/index.ts (hunk 3 above)
Comment (MartinMinkov) [reply frontend/snapps/httpsnapp/src/index.ts]: Added a way to pass in the prefix using some currying. The naming/usage feels a little off so please let me know if there are other improvements to be added. The usage can be seen in the `simple.test.ts` file.

**Pull request closed without merging**